datasync/kvdbsync/plugin_impl_dbsync.go

```diff
@@ -80,7 +80,27 @@ func (plugin *Plugin) Init() error {
 // The order of plugins in flavor is not important to resync
 // since Watch() is called in Plugin.Init() and Resync.Register()
 // is called in Plugin.AfterInit().
+//
+// If provided connection is not ready (not connected), AfterInit starts new goroutine in order to
+// 'wait' for the connection. After that, the new transport watcher is built as usual.
 func (plugin *Plugin) AfterInit() error {
+	if plugin.KvPlugin == nil || plugin.KvPlugin.Disabled() {
+		return nil
+	}
+	if plugin.KvPlugin.Connected() {
```

Comment:
I believe there is a race condition here. The `Connected` returns false and then before you call `OnConnect` to register the handler, it connects and the handler is never run.

One solution for this could be to check inside the `OnConnect` before adding handler to the slice if the connection is already established and run the function from there. Then the `Connected` method might actually be obsolete because you could just use `OnConnect`.

Reply:
you mean callback, not handler

Reply:
Sure, callback handler 🤣